Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -2493,7 +2491,11 @@ bool Document::is_fully_active() const
 
 bool Document::is_active() const
 {
-    auto navigable = this->navigable();
+    return is_active(navigable());
+}
+
+bool Document::is_active(JS::GCPtr<HTML::Navigable> navigable) const
+{
     return navigable && navigable->active_document() == this;
 }
```

Comment:
making `navigable` a parameter in this is unappropriate. that might only work if `is_active` is not a Document's method but a static function.

I think the better idea for now is to inline `navigable->active_document() == this;` whenever it saves from a navigable lookup.

Reply:
Any chance you could expand on why having both the navigable member function and a parameter here makes this unappropriate?

The downside of inlining the check is that `::update_layout` and `::is_active` both have an idea of what an active document is. IIRC `::is_active` wasn't possible.